Source/WebCore/rendering/RenderText.cpp

```diff
@@ -1477,9 +1477,10 @@ static inline bool isInlineFlowOrEmptyText(const RenderObject& renderer)
 
 UChar RenderText::previousCharacter() const
 {
-    // find previous text renderer if one exists
     const RenderObject* previousText = this;
     while ((previousText = previousText->previousInPreOrder())) {
+        if (!previousText->isInFlow())
+            continue;
```

Comment:
nit: this is slightly confusing as it implies there can be out-of-flow text (a box can be out-of-flow but that does not make the text inside it out-of-flow).
`while ((previousText = previousText->previousInPreOrder())) {`
should probably read
`while ((previousSibling = previousSibling->previousInPreOrder())) {`
again, just a nit.

Reply:
Yes, not something I introduced, but I agree the style in the function is confusing, with a variable named `previousText` that is not a text renderer. I hope you’re OK that I didn’t fix that. I don’t think the `isInFlow` check is particularly more confusing than the other things happening there, like checking if the `previousText` is a text renderer or not, so I will leave it as is for now.

Reply:
Absolutely. I am going to clean this up once your PR lands. I also noticed that the static helper we call here was incorrectly named `isInlineFlowOrEmptyText` -> `isInlineBoxOrEmptyText`

Reply:
I also wonder whether this logic is right for more unusual cases. The new code I added might be wrong if the text being capitalized is itself out of flow.

Reply:
I think this is more correct than not. Content inside an out-of-flow box is usually means it is "detached" from the rest of the flow (and in most cases it ends up somewhere else on the screen) and same applies to floating positioned content.